aten/src/ATen/native/cuda/Indexing.cu

```diff
@@ -0,0 +1,258 @@
+  }
+  int64_t nElemBefore, strideBefore, nElemAfter;
+  Tensor linearIndex;
+  std::tie(linearIndex, nElemBefore, strideBefore, nElemAfter) = computeLinearIndex(self, indices);
+  return std::make_tuple(linearIndex, self, nElemBefore, strideBefore, nElemAfter, inversePerm);
+}
+
+
+namespace {
+void index_put_accum_kernel(Tensor & self, TensorList indices, const Tensor & value) {
+  if (indices.size() > (size_t)self.dim()) {
+    AT_INDEX_ERROR("too many indices for tensor of dimension ", self.dim(), " (got ", indices.size(), ")");
+  }
+  auto value_ = value.contiguous();
+  Tensor linearIndex, expandedValue, src;
+  int64_t nElemBefore, strideBefore, sliceSize;
+  std::vector<int64_t> inversePerm;
+  std::tie(linearIndex, src, nElemBefore, strideBefore, sliceSize, inversePerm) = makeLinearIndex(self, indices);
+  int64_t num_indices = linearIndex.numel();
+  if (num_indices > 0 && sliceSize > 0) {
+      const bool permuted = !src.is_contiguous();
+      auto src_ = permuted ? src.contiguous() : src;
+      linearIndex = linearIndex.view(-1);
+      auto sorted_indices = at::empty_like(linearIndex);
+      auto orig_indices = at::empty_like(linearIndex);
+      using device_ptr = thrust::device_ptr<int64_t>;
+      const cudaStream_t stream = at::cuda::getCurrentCUDAStream();
+    
+      linearIndex.div_(sliceSize);
+      {
+      sorted_indices.copy_(linearIndex);
+      auto allocator = THCThrustAllocator(globalContext().lazyInitCUDA());
+      auto policy = thrust::cuda::par(allocator).on(stream);
+    
+      // Fill sortedOrigIndices with sequential indices
+      const auto count_iter = thrust::counting_iterator<int64_t>(0);
+      auto orig_data = device_ptr(orig_indices.data<int64_t>());
+      thrust::copy(policy, count_iter, count_iter + num_indices, orig_data);
+    
+      // Sort the inputs into sorted with the corresponding indices; we
+      // don't need a stable or multidimensional sort, so just use Thrust
+      // directly
+      // Sort; a stable sort is not required
+      // NB - not passing comparator causes thrust to use radix sort, and it hurts perf A LOT, at least for medium (few K) sized indices
+      auto sorted_data = device_ptr(sorted_indices.data<int64_t>());
+      thrust::sort_by_key(policy, sorted_data, sorted_data + num_indices, orig_data, ThrustLTOp<int64_t>());
+      }
+      TORCH_INTERNAL_ASSERT(linearIndex.numel()*sliceSize*nElemBefore == value.numel(), "number of flattened indices did not match number of elements in the value tensor", linearIndex.numel()*sliceSize*nElemBefore, value.numel());
+      TORCH_INTERNAL_ASSERT(self.numel() < std::numeric_limits<int>::max(), "index_put_ with accumulation is not supported on large tensors, number of source elements =", self.numel());
+      TORCH_INTERNAL_ASSERT(value.numel() < std::numeric_limits<int>::max(), "index_put_ with accumulation is not supported on large tensors, number of source elements =", value.numel());
```

Comment:
These should all be `TORCH_CHECK`, because a user in Python can write code that triggers these errors. (Remember: `TORCH_INTERNAL_ASSERT` is for any checks that would be OK to delete, because they *really really* should never be violated.)

Reply:
Oh, is this related to the removal of indexing checks you mentioned?